include/neml2/tensors/BatchTensor.h

```diff
@@ -73,6 +73,8 @@ class BatchTensor : public torch::Tensor
   /// Construct from existing tensor, no batch dimensions
   BatchTensor(const torch::Tensor & tensor);
 
+  BatchTensor(Real) = delete;
```

Comment:
There were unfortunately a lot of implicit conversions going on. I had to fix all of them.